### PR TITLE
#171146588 Fix user search functionality

### PR DESCRIPTION
**Pull request by victkarangwa**
#### What does this PR do?
- Fix search pagination issue
- Fix user search options
- Add search results counter
#### Description of Task to be completed?
- This PR fixes pagination issue when the user tries to search for a specific record
- This PR fixes user search option by allowing a user to search by any key relating to the record
- This PR also adds the counter for results retrieved according to the user search
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend.git`
- `cd team-odd-bn-frontend`
- `git checkout bg-fix-search-functionality-171146588`
- `npm install` packages
- `npm run dev`
-  Login to the system
- Go to the trips and then search
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#171146588](https://www.pivotaltracker.com/story/show/171146588)
#### Screenshots (if appropriate)
Screen Shot 2020-02-07 at 10 58 23
Screen Shot 2020-02-07 at 10 59 04
Screen Shot 2020-02-07 at 11 00 36

#### Questions:
N/A

### CODE REVIEW

File: src/assets/css/selectOption.scss [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 .selectOption {
-  height: 40px !important;
+  height: 30px !important;
   width: 100%;
   border: 1px solid $logo-color;
   outline: none;
   font-size: 14px;
   font-weight: bold;
-}
+}
```
Comment (hezronkimutai) [inline src/assets/css/selectOption.scss]: End of line

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
Hi, @victorkarangwa4 I left some feedback. Also, the tests are failing. Fix them and we will be good to go.

File: src/views/Dashboard/sidebar/TopRightSide.js [hunk 2]
```diff
@@ -1,133 +1,175 @@
...
+import '../../../assets/css/sidenav/sidebar.scss'
+import { Link } from 'react-router-dom'
+import notificationIcon from '../../../assets/images/notification_icon/notification_32px.png'
+import settingsIcon from '../../../assets/images/settings_icon/settings_32px.png'
+import defaultUserIcon from '../../../assets/images/default_user_icon/default_user_32px.png'
+import ToolTip from '../../../components/Sidebar/ToolTip'
+import initializeSocket from '../../../helpers/notificationHelper'
+import markAllAsRead from '../../../redux/actions/notifications/markAllAsReadActions'
+import markOneAsRead from '../../../redux/actions/notifications/markOneAsReadActions'
+import getAllNotifications from '../../../redux/actions/notifications/getAllNotificationActions'
```
Comment (NiyongaboEric) [inline src/views/Dashboard/sidebar/TopRightSide.js]: Why did you remove all semicolons?

File: src/views/trips/ManagerView.js [hunk 3]
```diff
@@ -56,25 +57,26 @@ const AvailRequestsView = ({
...
         className="search"
         id="searchInput"
         type="text"
         onChange={(e) => {
-          const newData = handleFilter(e, data, search.key);
+          const newData = handleFilter(e, data, search.key || 'manager.trips.any');
           searchResults(newData);
         }}
-        disabled={!search.key && 'disabled'}
+        // disabled={!search.key && 'disabled'}
```
Comment (NiyongaboEric) [inline src/views/trips/ManagerView.js]: Remove comment.

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]
Well done @victorkarangwa4. Please take a look at the feedback I left

**Review by nziokaivy — APPROVED** [review]
Good job Victor! Looks good to me!

**Review by NiyongaboEric — APPROVED** [review]

File: src/containers/accommodation/ViewAccommodations.js [hunk 4]
```diff
@@ -41,7 +41,9 @@ export class ViewAccommodations extends Component {
   }
 
   render() {
-    const { allLikesDislakes } = this.props;        
+    const { allLikesDislakes } = this.props;    
+    console.log('=====>', allLikesDislakes)
```
Comment (kipropbrian) [inline src/containers/accommodation/ViewAccommodations.js]: Why?

File: src/containers/admin/UserRoles.js [hunk 5]
```diff
@@ -4,20 +4,34 @@ import AllUsers from './allUsers';
...
     this.props.getAllUsers();
     this.props.getRolesAction();
   }
 
   componentDidUpdate() {
+    this.props.adminState.search.payload = undefined;
     this.props.adminState.admin.payload && this.props.getAllUsers();
     this.props.adminState.admin.payload = undefined;
+    if (this.props.adminState.allUsers.users && this.props.adminState.search.payload) {
+      const p = this.props.adminState.allUsers.users
```
Comment (kipropbrian) [inline src/containers/admin/UserRoles.js]: Destructure  all this props

File: src/containers/admin/UserRoles.js [hunk 6]
```diff
@@ -36,16 +51,22 @@ export class UserRoles extends Component {
           id="searchInput"
           type="text"
           onChange={(e) => {
-            const newData = handleFilter(e, data, 'email');
+            const newData = handleFilter(e, data);
             this.props.searchResults(newData);
-            return handleFilter(e, data, 'email');
+            this.props.searchInput(e.target.value.trimLeft());
```
Comment (kipropbrian) [inline src/containers/admin/UserRoles.js]: What about space on the right?

File: src/views/trips/RequestsView.js [hunk 7]
```diff
@@ -40,29 +78,41 @@ const RequestsView = ({
...
+            const newData = handleFilter(
+              e,
+              data,
+              search.key || 'user.trips.any',
+            );
+            searchInput(e.target.value.trimLeft());
             searchResults(newData);
           }}
-          disabled={!search.key && 'disabled'}
+          // disabled={!search.key && 'disabled'}
```
Comment (kipropbrian) [inline src/views/trips/RequestsView.js]: Remove if you are not using it

**Review by william000000 — APPROVED** [review]
Nice Job @victorkarangwa4

**Pull request merged**